nexia/home.py

```diff
@@ -182,39 +214,50 @@ async def post_url(self, request_url: str, payload: dict) -> aiohttp.ClientRespo
         response.raise_for_status()
         return response
 
-    async def _get_url(
+    async def get_url(
         self, request_url: str, headers: dict[str, str] | None = None
     ) -> aiohttp.ClientResponse:
         """
-        Returns the full session.get from the URL (ROOT_URL + url)
-        :param url: str
+        Returns the full session.get from the URL and headers
+        :param request_url: str
+        :param headers: headers to include in the get request
         :return: response
         """
         if not headers:
             headers = {}
 
         headers.update(self._api_key_headers())
-        _LOGGER.debug("GET: Calling url %s", request_url)
-        response = await self.session.get(
+        _LOGGER.debug("GET: Calling url %s with headers: %s", request_url, headers)
+
+        response: aiohttp.ClientResponse = await self.session.get(
             request_url,
             allow_redirects=False,
             timeout=TIMEOUT,
             headers=headers,
             max_redirects=MAX_REDIRECTS,
         )
-        _LOGGER.debug(
-            "GET: RESPONSE %s: response.status %s",
-            request_url,
-            response.status,
-        )
+        if self.log_response:
```

Comment:
Looks like duplicate code that could be moved into a function as its almost the same as above